### PR TITLE
Add e2e for customize query

### PR DESCRIPTION
**Pull request by bamader**
# PULL REQUEST

## Summary
This PR adds end to end testing for the customize query workflow. Various facets are tested, including selective valueset filtering, group checking/unchecking, and selecting/deselecting all.

## Related Issue
Fixes #https://github.com/CDCgov/dibbs-query-connector/issues/44

### CODE REVIEW

File: query-connector/e2e/customize_query.spec.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
+// @ts-check
+
+import { test, expect } from "@playwright/test";
+import { TEST_URL } from "../playwright-setup";
+import { PAGE_TITLES } from "@/app/query/stepIndicator/StepIndicator";
+
+import { TEST_PATIENT, TEST_PATIENT_NAME } from "./constants";
```
Comment (m-goggins) [inline query-connector/e2e/customize_query.spec.ts:7]: I'm not sure when this got added but definitely makes it easier to write tests!

File: query-connector/e2e/customize_query.spec.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    await expect(
+      page.getByRole("heading", { name: PAGE_TITLES["search"], exact: true }),
+    ).toBeVisible();
+
+    await page.getByRole("button", { name: "Fill fields" }).click();
+    await page.getByLabel("First Name").fill(TEST_PATIENT.FirstName);
+    await page.getByLabel("Last Name").fill(TEST_PATIENT.LastName);
+    await page.getByLabel("Date of Birth").fill(TEST_PATIENT.DOB);
+    await page.getByLabel("Medical Record Number").fill(TEST_PATIENT.MRN);
+    await page.getByLabel("Phone Number").fill(TEST_PATIENT.Phone);
```
Comment (m-goggins) [inline query-connector/e2e/customize_query.spec.ts:30]: Do we have to fill these fields if you already clicked "Fill Fields" on line 25?

File: query-connector/e2e/customize_query.spec.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        .getByRole("table")
+        .filter({ hasText: "I do not have housing" })
+        .getByRole("row"),
+    ).toHaveCount(6);
+    const acceptableSdohKeywords = [
+      "history",
+      "narrative",
+      "housing",
+      "pregnancy",
+      "with anonymous partner",
```
Comment (m-goggins) [inline query-connector/e2e/customize_query.spec.ts:176]: This might be out of scope for the ticket (or honestly over optimization at this stage), but what do you think about creating a customized query constant like we do for `TEST_PATIENT`? We could add key words like this as well as the expected medications, etc. I'm not sure if it's easier to update a constant or just update the tests at this point but thought I'd throw it out there.

**Review by m-goggins — APPROVED** [review]
Left a few non-blocking comments, but overall looks good!

File: query-connector/e2e/customize_query.spec.ts (hunk 1 above)
Comment (bamader) [reply query-connector/e2e/customize_query.spec.ts:7]: I thought the same thing! Was so glad to see it just appear haha. Might be harder to use once we support multiple default or test patients like we used to, but for now it's absolutely great.

File: query-connector/e2e/customize_query.spec.ts (hunk 2 above)
Comment (bamader) [reply query-connector/e2e/customize_query.spec.ts:30]: Good question 🤔 . I just copied this setup stuff from the initial part of the regular query_spec tests, but I'd imagine not actually. As long as that test patient remains the same we shouldn't need them.

**Review by robertandremitchell — APPROVED** [review]
looks good!

i imagine some of the stuff at the beginning, like navigating to the searchform page, is carry-over from prior tests. I wonder if we need it here, or if we could just directly to the /query endpoint. If/when we do start to change the copy of the home page, it'll just mean having to update this, when in theory it has nothing to really do with our customize query code.

I also ran into some issues running the tests locally, but this seems to be a known confusing issue, so 🤷🏽

File: query-connector/e2e/customize_query.spec.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        .filter({ hasText: "doxycycline hyclate 100 MG" })
+        .getByRole("row"),
+    ).toHaveCount(2);
+  });
+
+  test("customize query select / deselect all filters whole DibbsConceptType, across tabs", async ({
+    page,
+  }) => {
+    // await expect(
+    //   page.getByText("249 labs found, 4 medications found, 104 conditions found."),
```
Comment (fzhao99) [inline query-connector/e2e/customize_query.spec.ts:108]:
```suggestion
```
did we mean to clean this comment up?

File: query-connector/e2e/query_workflow.spec.ts [hunk 5]
```diff
@@ -62,23 +62,27 @@ test.describe("querying with the Query Connector", () => {
...
-    ).not.toBeVisible();
-
-    await page.getByText("Return to Select query").click();
+    // Update as the unlucky dev: Chromium now has issues with this check present
+    // for chlamydia, as well as issues in another test checking for non-zero values.
+    // Other browsers have no problem navigating both checks as they stand.
+    // Update to the update: the connection issue has broadened to include webkit
+    // and seems to apply with no rhyme or reason to use case. Commenting this out
+    // because it no longer acts as a sanity check but leaving it here as a warning
+    // to others.
```
Comment (fzhao99) [inline query-connector/e2e/query_workflow.spec.ts:77]: 😭

**Review by fzhao99 — APPROVED** [review]
lgtm! sucks that you ran into whatever playwright edge case that bit me a few weeks ago, but glad things are working in CI

File: query-connector/e2e/customize_query.spec.ts (hunk 4 above)
Comment (bamader) [reply query-connector/e2e/customize_query.spec.ts]: Yep, cleaned
